integration/stability/soak_parallel_rm.sh

```diff
@@ -0,0 +1,198 @@
+#!/bin/bash
+# Copyright (c) 2017 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This test will run a number of parallel containers, and then try to
+# 'rm -f' them all at the same time. It will check after each run and
+# rm that we have the expected number of containers, shims, proxys,
+# qemus and runtimes active
+# The goals are two fold:
+# - spot any stuck or non-started components
+# - catch any hang ups
+
```

Comment:
Might be good to have a `set -e` in here. Note that I checked with that locally, but the script then exited prematurely so I had to change a few of the functions to add `|| true` and also this...

```
((how_many++))
```

... had to change to:

```
how_many=$((how_many + 1))
```

Reply:
hmmm. I'm a bit wary of setting `-e` here - if then docker or something fails because we have a bug then the script would just quit, without a chance of dumping any diagnostics. I think I'd rather have the script find that something 'had not happened', and then give it the chance to dump out the diagnostics.
Let me push an update to the script btw, which checks some other things (that `cc-runtime list` and /var/lib/virtcontainers/pods both have the correct number of items - and also I might have the 'check number of mounts' in as well - but I need to stare at that a bit)... hold on, I'll do a push - but, be warned, with the runtime as it is at the minute the test will fail on the vc/pods check I believe)

Reply:
Given what this script is doing, yeah that's a fair point.